src/components/ShareDialog.jsx

```diff
@@ -36,51 +36,55 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-export default function ShareDialog(props) {
+export default function ShareDialog() {
+  const location = useLocation();
+  const { showShareDialog, shareTitle, shareURL } = location.state
+    ?.shareDialog || {
+    showShareDialog: false,
+  };
```

Comment:
Using the other suggested format in my other comment I think this can be changed to:
```jsx
  const showShareDialog = location.state.showShareDialog ?? false;
  const { text, title, url } = location.state.shareData ?? {};
```